models/gif_psc_exp.cpp

```diff
@@ -0,0 +1,440 @@
+/*
+ *  gif_psc_exp.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "gif_psc_exp.h"
+
+// C++ includes:
+#include <limits>
+
+// Includes from nestkernel:
+#include "exceptions.h"
+#include "kernel_manager.h"
+#include "universal_data_logger_impl.h"
+
+// Includes from sli:
+#include "dict.h"
+#include "integerdatum.h"
+#include "doubledatum.h"
+#include "dictutils.h"
+
+#include "numerics.h"
+#include "compose.hpp"
+#include "propagator_stability.h"
+
+namespace nest
+{
+/* ----------------------------------------------------------------
+ * Recordables map
+ * ---------------------------------------------------------------- */
+
+RecordablesMap< gif_psc_exp > gif_psc_exp::recordablesMap_;
+
+// Override the create() method with one call to RecordablesMap::insert_()
+// for each quantity to be recorded.
+template <>
+void
+RecordablesMap< gif_psc_exp >::create()
+{
+  // use standard names wherever you can for consistency!
+  insert_( names::V_m, &gif_psc_exp::get_V_m_ );
+  insert_( names::E_sfa, &gif_psc_exp::get_E_sfa_ );
+  insert_( names::I_stc, &gif_psc_exp::get_I_stc_ );
+  insert_( names::input_currents_ex, &gif_psc_exp::get_input_currents_ex_ );
+  insert_( names::input_currents_in, &gif_psc_exp::get_input_currents_in_ );
```

Comment:
Well, this now needs to change because of issue #501 and PR #502: can you replace ``input_currents_ex/in`` by ``I_syn_ex/in``? ``input_currents_ex`` is going to disappear from ``nest_names``.